src/sentry/eventstream/snuba.py

```diff
@@ -157,32 +161,37 @@ def insert(
 
         is_transaction_event = self._is_transaction_event(event)
 
+        extra_data: List[Mapping[Any, Any]] = [
+            {
+                "group_id": event.group_id,
+                "group_ids": [group.id for group in event.groups],
```

Comment:
Curious why you're keeping the group ids in here and the group states relating to these groups in another place? Did you consider keeping it all together?

Reply:
@wedamija it looks like `group_ids` in the message body isn't used? Can we safely delete this?

Reply:
This is definitely used in the snuba consumer to write these ids to clickhouse. We can consolidate this into a dict if we want though, but you will need to make changes to the snuba consumer as well https://github.com/getsentry/snuba/blob/ea526f5af8b8405c814d47592793081c1c767b70/snuba/datasets/transactions_processor.py#L139-L141

Reply:
Oh wait, this is in the message body right? If we're going to store the group ids in the headers then we don't really need to consolidate

Reply:
Ahh, I see so `group_ids` is used in snuba to store to clickhouse. But the new `group_states` in the header will only be used in `post_process_forwarder`.